### PR TITLE
feat(dedicated): remove delete action for BYOIP

### PR DESCRIPTION
**Pull request by mhelhali-soufien**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-14605
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

### CODE REVIEW

**Review by sachinrameshn — DISMISSED** [review]

File: packages/manager/apps/dedicated/client/app/ip/components/list/byoip/delete/ip-ip-byoip-delete.controller.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export default /* @ngInject */ ($scope, coreURLBuilder) => {
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/ip/components/list/byoip/delete/ip-ip-byoip-delete.controller.js:1]:
```suggestion
export default class IpByoipDeleteController {
/* @ngInject */ 
constructor ($scope, coreURLBuilder) {
  this.$scope = $scope;
  this.coreURLBuilder = coreURLBuilder;
}
```

File: packages/manager/apps/dedicated/client/app/ip/components/list/byoip/delete/ip-ip-byoip-delete.controller.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  $scope.data = $scope.currentActionData;
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/ip/components/list/byoip/delete/ip-ip-byoip-delete.controller.js:2]:
```suggestion
  this.data = $scope.currentActionData;
```
use this inside the controller

File: packages/manager/apps/dedicated/client/app/ip/components/list/byoip/delete/ip-ip-byoip-delete.controller.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  $scope.serviceLink = coreURLBuilder.buildURL(
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/ip/components/list/byoip/delete/ip-ip-byoip-delete.controller.js:3]:
```suggestion
  this.serviceLink = coreURLBuilder.buildURL(
```

File: packages/manager/apps/dedicated/client/app/ip/components/list/byoip/delete/ip-ip-byoip-delete.controller.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    'dedicated',
+    '#/billing/autorenew',
+  );
+  $scope.cancelAction = () => {
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/ip/components/list/byoip/delete/ip-ip-byoip-delete.controller.js:7]:
```suggestion
  this.cancelAction = () => {
```

File: packages/manager/apps/dedicated/client/app/ip/components/list/byoip/delete/ip-ip-byoip-delete.controller.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    $scope.resetAction();
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/ip/components/list/byoip/delete/ip-ip-byoip-delete.controller.js:8]:
```suggestion
    this.resetAction();
```

File: packages/manager/apps/dedicated/client/app/ip/components/list/byoip/delete/ip-ip-byoip-delete.html [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<div data-ng-controller="IpDeleteByoipCtrl">
+    <oui-modal
+        heading="{{:: 'ip_byoip_delete_block_title' | translate }}"
+        type="warning"
+        on-dismiss="cancelAction()"
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/ip/components/list/byoip/delete/ip-ip-byoip-delete.html:5]:
```suggestion
        on-dismiss="IpDeleteByoipCtrl.cancelAction()"
```

File: packages/manager/apps/dedicated/client/app/ip/components/list/byoip/delete/ip-ip-byoip-delete.html [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+    >
+        <p
+            data-translate="ip_byoip_delete_block_description"
+            data-translate-values="{ link: serviceLink }"
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/ip/components/list/byoip/delete/ip-ip-byoip-delete.html:9]:
```suggestion
            data-translate-values="{ link: IpDeleteByoipCtrl.serviceLink }"
```

**Review by sachinrameshn — APPROVED** [review]

**Review by qpavy — DISMISSED** [review]

**Review by chipp972 — APPROVED** [review]

**Review by qpavy — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

**Pull request merged**